uPortal-webapp/src/main/webapp/WEB-INF/flows/edit-portlet/portletConfig.jsp

```diff
@@ -478,52 +473,35 @@ PORTLET DEVELOPMENT STANDARDS AND GUIDELINES
                                                 <c:set var="principal_name" value="${fn:escapeXml(principal.name)}" />
                                                 <c:set var="principal_browse_perm" value="${principal.typeAndIdHash}_BROWSE" />
                                                 <c:set var="principal_subscribe_perm" value="${principal.typeAndIdHash}_SUBSCRIBE" />
-                                                <c:if test="${supportsConfig}">
-                                                    <c:set var="principal_configure_perm" value="${principal.typeAndIdHash}_PORTLET_MODE_CONFIG" />
-                                                </c:if>
+                                                <c:set var="principal_configure_perm" value="${principal.typeAndIdHash}_PORTLET_MODE_CONFIG" />
                                                 <tr>
                                                     <td><i class="fa fa-users"></i> ${principal_name}</td>
-                                                    <c:choose>
-                                                      <c:when test="${portlet.permissions.contains(principal_browse_perm) || !portlet.permissions.contains(principal_subscribe_perm)}">
-                                                        <td class=text-center><input type="checkbox" name="${principal_browse_perm}" checked></td>
-                                                      </c:when>
-                                                      <c:otherwise>
-                                                        <td class=text-center><input type="checkbox" name="${principal_browse_perm}"></td>
-                                                      </c:otherwise>
-                                                    </c:choose>
-                                                    <c:choose>
-                                                      <c:when test="${portlet.permissions.contains(principal_subscribe_perm) || !portlet.permissions.contains(principal_browse_perm)}">
-                                                        <td class=text-center><input type="checkbox" name="${principal_subscribe_perm}" checked></td>
-                                                      </c:when>
-                                                      <c:otherwise>
-                                                        <td class=text-center><input type="checkbox" name="${principal_subscribe_perm}"></td>
-                                                      </c:otherwise>
-                                                    </c:choose>
-                                                    <c:if test="${supportsConfig}">
-                                                        <c:choose>
-                                                            <c:when test="${portlet.permissions.contains(principal_configure_perm) || !portlet.permissions.contains(principal_browse_perm)}">
```

Comment:
The main issue/bug with the PR before commit #3 was with this line (above).

It's not a good idea to determine `checked` based on `portlet.permissions.contains(principal_configure_perm) || !portlet.permissions.contains(principal_browse_perm)` -- what does the second part of the test have to do with it???

I realize you were just following the pattern from the previous 2 `<td>`s, but those weren't a good idea either.

What the previous 2 `test` expressions were trying to accomplish was:  "if I either have the permission explicitly, or if I don't have any permissions at all, default this checkbox to checked."

It was a hackish way of tricking the jsp into checking both checkboxes for a new principal (since principals with no checkboxes checked are removed with any page submit).  The approach was brittle -- your update would have broken it... the hack (as it was) only works if there are only 2 permissions in the form.

Reply:
Thanks @drewwills - that is a cleaner approach.